Content.Server/GameTicking/Rules/Components/NinjaRuleComponent.cs

```diff
@@ -0,0 +1,59 @@
+using Content.Server.Ninja.Systems;
+using Content.Server.Objectives;
+using Robust.Shared.Audio;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+
+namespace Content.Server.GameTicking.Rules.Components;
+
+[RegisterComponent, Access(typeof(NinjaSystem))]
+public sealed class NinjaRuleComponent : Component
+{
+    /// <summary>
+    /// List of objective prototype ids to add
+    /// </summary>
+    [DataField("objectives", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<ObjectivePrototype>))]
+    public readonly List<string> Objectives = new();
+
+    // TODO: move to job and use job???
+    /// <summary>
+    /// List of implants to inject on spawn
+    /// </summary>
+    [DataField("implants", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<EntityPrototype>))]
+    public readonly List<string> Implants = new();
+
+    /// <summary>
+    /// List of threats that can be called in
+    /// </summary>
+    [DataField("threats", required: true)]
+    public readonly List<Threat> Threats = new();
```

Comment:
I think threats would make more sense as part of ninjacomponent

Reply:
codewise probably easier for comms to handle it yeah
dunno if malf should be able to summon dragons and what not but at least it would be possible

Reply:
Well i'll leave it up to you to move em to the ninja or the comms console then

Reply:
ill just leave it how it is then make new pr to put threats on comms console component and add a nice api for it